ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useCreateReport.ts

```diff
@@ -25,7 +26,39 @@ const defaultResult = {
     error: null,
 };
 
+function trackReportCreation(
+    analyticsTrack: ReturnType<typeof useAnalytics>['analyticsTrack'],
+    reportConfiguration: ReportConfiguration
+) {
+    const { severities, fixability, imageTypes } = reportConfiguration.vulnReportFilters;
+    const { notifiers } = reportConfiguration;
+
+    const hasEmailNotifier =
+        notifiers.length > 0 && notifiers.some((notifier) => notifier.emailConfig);
```

Comment:
Good question, I think my thought process here was that if the notifier attached to a report config ever supported a different type, and `emailConfig` became optional, that this would throw a compile error. Just checking the length didn't feel like enough to distinguish that a report `hasEmailNotifier`.

e.g. If the current:
```typescript

export type ReportConfiguration = {
    id: string;
    name: string;
    description: string;
    type: ReportType;
    vulnReportFilters: VulnerabilityReportFilters;
    notifiers: NotifierConfiguration[];
    schedule: Schedule | null;
    resourceScope: ResourceScope;
};

export type NotifierConfiguration = {
    emailConfig: {
        notifierId: string;
        mailingLists: string[];
        customSubject: string;
        customBody: string;
    };
    notifierName: string;
};
```
was changed to:
```ts

export type ReportConfiguration = {
    id: string;
    name: string;
    description: string;
    type: ReportType;
    vulnReportFilters: VulnerabilityReportFilters;
    notifiers: (EmailConfiguration | StorageConfiguration)[];
    schedule: Schedule | null;
    resourceScope: ResourceScope;
};

export type EmailConfiguration = {
    emailConfig: {
        notifierId: string;
        mailingLists: string[];
        customSubject: string;
        customBody: string;
    };
    notifierName: string;
};

export type StorageConfiguration = {
    storageConfig: {
        ...
    };
    notifierName: string;
};
```
With the property check we would get a TS error on this line, since `emailConfig` isn't always valid on the expected type. Without the check, analytics events would be tracked as long as there was some notifier, even if it wasn't an email notifier.

I've seen in the past that analytics can easily get out of sync, and there isn't really a good indication of this until a lot of bad data ends up in the system, so I tried to be a little overly cautious here.